### PR TITLE
`General`: Verify that all courses have a specified accuracy of scores

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).

### Motivation and Context

Old courses did not have a specified accuracy of scores which can lead to problems in the UI of Artemis as well as the scores of student submissions.

### Description

This PR makes the column for the accuracy of scores to be non nullable and set the value to 1 for all null values at the moment.
To verify that, this PR makes the field for the accuracy of scores in the course edit/create page a required field and the server checks that this field is set in the Rest call.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Administrator

1. Edit an existing (preferably older than 3-4 months) and check that the accuracy of scores has to be set in the form
2. Create a new course and also check that the accuracy of scores has to be set in the form
3. Check that after creating/editing the value is correctly stored. (Reenter the edit page and check the current value)

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by daniels98it — DISMISSED** [review]
Tested in testing session and works as expected 👍

**Review by FlorianGlombik — APPROVED** [review]
Tested on ts1 with an old course from SS20 - I did not find something that did not work, the accuracy was set to the default value 1.
The `400` was also returned when sending an invalid request without setting the accuracy.

The code looks fine aswell.

File: src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java [hunk 1]
```diff
@@ -388,6 +390,12 @@ private void validateOnlineCourseAndRegistrationEnabled(Course course) {
         }
     }
 
+    private void validateAccuracyOfScores(Course course) {
+        if (course.getAccuracyOfScores() == null) {
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java:394]: I think we should also check that the value makes sense and exclude negativ values and values larger than 5 or 10

**Review by krusche — CHANGES_REQUESTED** [review]
I added some small comments, should be easy to fix

File: src/main/webapp/app/course/manage/course-update.component.ts [hunk 2]
```diff
@@ -45,7 +45,6 @@ export class CourseUpdateComponent implements OnInit {
     complaintsEnabled = true; // default value
     requestMoreFeedbackEnabled = true; // default value
     customizeGroupNames = false; // default value
-    accuracyOfScores = 1; // default value
```
[the remark this answers is not in the document]
Comment (JohannesStoehr) [reply src/main/webapp/app/course/manage/course-update.component.ts:48]: This is already set in course.model.ts when creating a new course.
This variable in course-update.component.ts was completely unused

[the remark this answers is not in the document]
Comment (krusche) [reply src/main/webapp/app/course/manage/course-update.component.ts:48]: great, thanks for the feedback

**Review by krusche — APPROVED** [review]

**Pull request merged**